### PR TITLE
Implement Privacy Provider

### PR DESCRIPTION
**Pull request by good-moon-oooo**

### CODE REVIEW

**Comment by keevan** [comment]
Changes look OK from a quick look.

Have you run unit tests to make sure everything looks OK? I realise the plugin doesn't have any CI so it's somewhat hard to tell if it works or is broken.

Also, for naming conventions, should we target master-catalyst (since master appears to be in the same state) instead?

**Comment by good-moon-oooo** [comment]
HI @keevan , thanks for reviewing. Yep, the tests have been run locally and everything is ok. 

For the branch name, thanks for the advice. I will change it to master-catalyst.

File: classes/privacy/provider.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class provider implements \core_privacy\local\metadata\null_provider {
+    /**
+     * Get the language string identifier with the component's language
+     * file to explain why this plugin stores no data.
+     *
+     * @return  string
+     */
+    public static function get_reason() : string {
```
Comment (keevan) [inline classes/privacy/provider.php]:
```suggestion
    public static function get_reason(): string {
```

**Review by keevan — APPROVED** [review]
One more small thing, but otherwise happy for it to go in.

File: classes/privacy/provider.php (hunk 1 above)
Comment (good-moon-oooo) [reply classes/privacy/provider.php]: ah, thanks for pointing that out. I will squash this change into the commit and then use code_checker to check the code style of this commit.

Comment (good-moon-oooo) [reply classes/privacy/provider.php]: HI @keevan, the style fix is done, and also code checker is used. Everything ok this time.

**Pull request merged**